### DISCUSSION TITLE
does Delta reader project columns

### DISCUSSION THREAD

**Discussion by djouallah** [delta-io/delta-rs#2471] [General]
I am using this read a list of files from a delta table, but based on memory usage, it seems the reader is loading the whole table first ? is there a better way to read only a specific column?

`existing_files = DeltaTable(Destination,storage_options=storage_options).to_pandas(columns=["file"])['file'].unique().tolist()`

**Comment by ion-elgreco** [comment]
> I am using this read a list of files from a delta table, but based on memory usage, it seems the reader is loading the whole table first ? is there a better way to read only a specific column?
> 
> `existing_files = DeltaTable(Destination,storage_options=storage_options).to_pandas(columns=["file"])['file'].unique().tolist()`

To_pandas materializes the whole table. You should project on the pyarrow_dataset and then collect.

**Reply by djouallah** [reply]
Thanks

**Comment by djouallah** [comment]
this looks a bit too verbose :)

`DeltaTable(Destination,storage_options=storage_options).to_pyarrow_dataset().to_table(columns=['file']).to_pandas()['file'].unique().tolist()`

**Reply by ion-elgreco** [reply]
`pl.scan_delta(destination, storage_options=storage_options).select('file').collect().get_column('file')`
